File: SWEA/D3/2805. 농작물 수확하기/농작물 수확하기.py
T = int(input())
for test_case in range(1, T + 1):
    N = int(input())
    farm = []
    for _ in range(N):
        farm.append(list(map(int, input())))
    profit = 0
    s = N // 2
    e = N // 2
    for i in range(N):
        for j in range(s, e + 1):
            profit += farm[i][j]
        if i < N // 2:
            s -= 1
            e += 1
        else:
            s += 1
            e -= 1
    print(f'#{test_case} {profit}')

File: SWEA/D3/2805. 농작물 수확하기/농작물 수확하기.java
import java.util.Scanner;

class Solution
{
	public static void main(String args[]) throws Exception
	{
		Scanner sc = new Scanner(System.in);
		int T=sc.nextInt();
		
		for(int test_case = 1; test_case <= T; test_case++)
		{
            int N = sc.nextInt();
            int[][] farm = new int[N][N];
            
            int answer = 0;
            
            for(int i = 0; i < N; i++)
            {
                String s = sc.next();
                for(int j = 0; j < N; j++)
                {
                	farm[i][j] = s.charAt(j) - '0';
                    if(i == N / 2)
                        answer += farm[i][j];
                }
            }
            
            for(int i = 0; i < N / 2; i++)
            {
            	for(int j = N / 2 - i; j <= N / 2 + i; j++)
                {
                	answer += farm[i][j];
               	    answer += farm[N - 1 - i][j];
                }
            }
            
            System.out.printf("#%d %d\n", test_case, answer);
		}
	}
}